### PR TITLE
Upload receipts directly from Receipts page

### PR DESCRIPTION
**Pull request by cjdenio**
Adds the ability to upload receipts from the Receipts page, both via a button and drag-and-drop.

https://user-images.githubusercontent.com/34525547/167277358-77b23455-17d9-4980-80b8-6896ed272bf6.mov

Closes #1996!

### CODE REVIEW

File: app/assets/stylesheets/components/_physical_cards.scss [hunk 1]
```diff
@@ -98,6 +98,8 @@
 }
 
 .stripe-card {
+  overflow: hidden;
```
Comment (maxwofford) [inline app/assets/stylesheets/components/_physical_cards.scss]: What's this used for?

File: app/controllers/hcb_codes_controller.rb [hunk 2]
```diff
@@ -44,7 +44,11 @@ def receipt
       ::HcbCodeService::Receipt::Create.new(attrs).run
     end
 
-    redirect_to params[:redirect_url], flash: { success: "Receipt".pluralize(params[:file].length) + " added!" }
+    if params[:show_link]
+      redirect_to params[:redirect_url], flash: { success: { text: "Receipt".pluralize(params[:file].length) + " added!", link: hcb_code_path(@hcb_code), link_text: "View" } }
```
Comment (maxwofford) [inline app/controllers/hcb_codes_controller.rb]: Could you add a space to this before "View"?

File: app/views/application/_flash.html.erb [hunk 3]
```diff
@@ -4,7 +4,14 @@
       role="alertdialog"
       class="flash flex items-center mt3 mb2 mx-auto shrink-none <%= key %>"
       data-behavior="flash">
-      <span class="flex-auto"><%= value %></span>
+      <span class="flex-auto">
+        <% if value.is_a? Hash %>
+          <%= value["text"] %>
+          <%= link_to value["link_text"], value["link"] if value["link"] %>
```
Comment (maxwofford) [inline app/views/application/_flash.html.erb]: Maybe we could add an icon here? Something just to make the "view" link a little more visually distinct.

**Review by maxwofford — APPROVED** [review]
This looks great! Left a couple comments to think about, but I'm happy with this merged in as is.

File: app/assets/stylesheets/components/_physical_cards.scss (hunk 1 above)
Comment (cjdenio) [reply app/assets/stylesheets/components/_physical_cards.scss]: Oh, that just fixes an issue with the hover effect on Stripe cards:

Screen Shot 2022-05-08 at 1 43 59 PM

File: app/controllers/hcb_codes_controller.rb (hunk 2 above)
Comment (cjdenio) [reply app/controllers/hcb_codes_controller.rb]: Hmm, what do you think about right-aligning the link instead?

(Sort of like Material Design snackbars)

Screen Shot 2022-05-08 at 2 24 44 PM

**Review by exu3 — COMMENTED** [review]
:ohyeahwoofyeah: this is great!

Small thing I noticed—there's some horizontal scroll on transactions with longer memos.

**Comment by cjdenio** [comment]
Reviewed with @maxwofford over Zoom!

**Pull request merged**